### PR TITLE
Add type annotations for aiven.client.argx

### PR DESCRIPTION
**Pull request by rominf**
# About this change: What it does, why it matters

Partial implementation of #258.

### CODE REVIEW

**Comment by rominf** [comment]
Depends on #273

File: aiven/client/argx.py [hunk 1]
```diff
@@ -193,50 +200,60 @@ def add_cmds(self, add_func):
...
 
+    def _to_mapping_collection(
+        self, obj: Union[Mapping[str, Any], Collection[Mapping[str, Any]]], single_item: bool = False
+    ) -> Collection[Mapping[str, Any]]:
+        if single_item:
+            assert isinstance(obj, Mapping)
+            return [obj]
+        else:
+            assert isinstance(obj, Collection)
+            return cast(Collection[Mapping[str, Any]], obj)
```
Comment (dmitrii-vasilev) [inline aiven/client/argx.py:227]: Is it possible to avoid `cast` here? The `assert` on the previous line suppose to limit the result to only one option.

**Review by dmitrii-vasilev — DISMISSED** [review]
Some comments

File: aiven/client/argx.py (hunk 1 above)
Comment (rominf) [reply aiven/client/argx.py:227]: Unfortunately, mypy is not that smart:
```
aiven/client/argx.py:227: error: Incompatible return value type (got "Union[Mapping[str, Any], Collection[Mapping[str, Any]]]", expected "Collection[Mapping[str, Any]]")
```

Comment (dmitrii-vasilev) [reply aiven/client/argx.py:228]: Hopefully we can fix it later

**Review by dmitrii-vasilev — APPROVED** [review]
LGTM

**Pull request merged**